feat(issue-priority): Update default search to filter out low priority issues

snigdhas opened the pull request:
BE counterpart to https://github.com/getsentry/sentry/pull/65211

snigdhas committed:
Update default search to filter out priority

malwilley commented on src/sentry/api/helpers/group_index/index.py:
```diff
@@ -81,6 +81,9 @@ def build_query_params_from_request(
     has_query = request.GET.get("query")
     query = request.GET.get("query", "is:unresolved").strip()
+    if features.has("organizations:issue-priority-ui", organization):
+        query = f"{query} issue.priority:[high, medium]"
```
I don't think this is quite right. We only want this to be used when `query` is None, but this looks like it will append the issue priority value to it even if we pass in a custom value

snigdhas committed:
Update default query

JoshFerge reviewed (dismissed):
This behavior needs tests -- also, are we removing `is:unresolved"` from the default query now?

scttcper commented on src/sentry/api/helpers/group_index/index.py:
```diff
@@ -80,7 +80,15 @@ def build_query_params_from_request(
+    query = request.GET.get("query", None)
+    if query is None:
+        if features.has("organizations:issue-priority-ui", organization):
+            query = f"{query} issue.priority:[high, medium]"
```
this looks like it would be "None issue.priority:[high, medium]"

wedamija commented on src/sentry/api/helpers/group_index/index.py:
```diff
@@ -80,7 +80,15 @@ def build_query_params_from_request(
             raise ParseError(detail="Invalid cursor parameter.")
     has_query = request.GET.get("query")
-    query = request.GET.get("query", "is:unresolved").strip()
+    query = request.GET.get("query", None)
+    if query is None:
```
We should probably also handle an empty string query here

malwilley replied:
Actually @snigdhas I think the existing behavior was correct? We still need to allow users to search without any filters, which would be the empty search string

snigdhas replied:
lol smart 🧠

snigdhas committed:
Handle empty strings

snigdhas committed:
Add test for priority default search

snigdhas committed:
Merge branch 'master' into snigdha/priority-default-be

wedamija committed:
feat(crons-db): Stop querying `MonitorEnvironment.environment` in queries (#65368)

We want to move all crons related tables to a separate database. To
facilitate this, we need to stop filtering to Environment using a join
and instead fetch ids from Environment in a separate query.

leeandher committed:
fix(commit-context): Update activity text (#65473)

Before:

<img width="544" alt="image"
src="https://github.com/getsentry/sentry/assets/35509934/a5e60091-8392-4890-88b9-23017a75503b">


After:
<img width="405" alt="image"
src="https://github.com/getsentry/sentry/assets/35509934/86571ce5-7e5c-44e7-bcf5-134fa7b2bb22">

michellewzhang committed:
fix(feedback): check and filter if contexts feedback message is none (#65479)

Fixes SENTRY-2NH5

- check if a feedback is missing `contexts`, `contexts[feedback]`, or
`contexts[feedback][message]`
- if so, filter it out first

anonrig committed:
build: disable biome formatter for now (#65464)

snigdhas committed:
Allow empty string queries

michellewzhang committed:
ref(replays): update user action breadcrumb color (#65485)

Per this
[figma](https://www.figma.com/file/uYS5xCq0m3SjtLKlxl7ilC/Specs%3A-Component-Names-in-Replay-Detail?type=design&node-id=0-1&mode=design&t=boX77bWp0pvCtnAh-0),
we're changing all the user action breadcrumb items to be blue in
preparation for react component names:

<img width="583" alt="SCR-20240220-nsto"
src="https://github.com/getsentry/sentry/assets/56095982/17477bc2-c1ef-4a15-ba12-a0e51c513cba">
<img width="588" alt="SCR-20240220-nrcv"
src="https://github.com/getsentry/sentry/assets/56095982/f5835bbd-bcb0-4065-b0ef-516209a9ca4f">

cathteng committed:
chore(slack): log entire slack unfurl error (#65471)

Abdkhan14 committed:
fix(trace-view-warnings): Added new svg. (#65487)

Changed image for banner:
<img width="1502" alt="Screenshot 2024-02-20 at 6 42 36 PM"
src="https://github.com/getsentry/sentry/assets/60121741/5b0a00b5-681e-4400-a736-80ff83568367">

Co-authored-by: Abdullah Khan <abdullahkhan@PG9Y57YDXQ.local>

wedamija committed:
feat(crons-db): Break foreign key on `MonitorEnvironment.environment` (#65370)

This removes the fk from `MonitorEnvironment.environment` and the
relationship from Django

JoshFerge committed:
fix(issues): allow additional properties on issues platform event schema (#65493)

set additionalProperties to true on the issue platform JSON validation
schema to be more lax, which should help with upstream changes where
more fields are added to events.

Related to #inc-646

JoshFerge commented on tests/snuba/api/endpoints/test_organization_group_index.py:
```diff
@@ -1924,6 +1924,34 @@ def test_filter_not_unresolved(self):
         assert response.status_code == 200
         assert [int(r["id"]) for r in response.data] == [event.group.id]
 
+    def test_default_search(self):
```
could we add a resolved issue here and make sure that it doesn't show up in the default? or is there already one set up by the test harness

mrduncan committed:
ref(issues): Mark team endpoints private (#65468)

These are only used on a single table within the stats page and will
potentially be changed in the not too distant future, also confirmed all
have extremely low usage.

JoshFerge commented on src/sentry/api/helpers/group_index/index.py:
```diff
@@ -80,8 +80,17 @@ def build_query_params_from_request(
-    query = request.GET.get("query", "is:unresolved").strip()
+    query = request.GET.get("query", None)
+    if query is None:
+        if features.has("organizations:issue-priority-ui", organization):
+            query = "is:unresolved issue.priority:[high, medium]"
+        else:
+            query = "is:unresolved"
```
nit: it could be good for code organization to [move](https://refactoring.com/catalog/extractFunction.html) this to a helper method, like `_get_default_query`, but your call on that

scttcper committed:
chore(ui): Fix formatting error in releaseChart (#65497)

Commit:
chore(grouping): fix typo (#65503)

Dav1dde committed:
feat(relay): Add bucket encoding global option (#65410)

snigdhas committed:
Restructure code and check resolved issues in tests

Commit:
feat(ddm): Add instrumentation for query timerange (#65425)

obostjancic committed:
fix(ddm): add to dashboard modal (#65519)

obostjancic committed:
feat(dashboards): multi query widget import (#65523)

ArthurKnaus committed:
chore(ddm): Add formulas feature flag (#65521)

anonrig committed:
feat: add typings for web/client_config.py (#65418)

Adds the typings for a new file.

obostjancic committed:
chore: prettier ignore (#65527)

AbhiPrasad committed:
feat(node): Use CJS require in onboarding (#65460)

Given our current troubles with ESM, let's put the CJS snippets first
for Node onboarding. We can change this later once v8 comes out.

Commit:
ref(dynamic-sampling): Improve dynamic sampling code (#65520)

Zylphrex committed:
perf(metrics): Move span examples query conditions around (#65467)

Snuba will rewrite the condition into

```sql
PREWHERE in(span_id, [16435934])
```

if we write the SnQL like `WHERE in(span_id, ['facade'])`. Note that it
has to be written specifically like this for Snuba to properly optimize
the query.

leeandher committed:
ref(auto-assignment): Invalidate assignee cache after project ownership changes (#65281)

I believe this will solve the caching issue since if the GroupOwner was
created via ownership, it may no longer be valid.

Resolves https://github.com/getsentry/sentry/issues/48564

obostjancic committed:
chore(alerts): remove migration ui flag (#65526)

markstory committed:
feat(features) Add helpers for option based rollouts (#65445)

We have a few variants of this style of logic in the application today.
If folks are happy with these functions I'll update the existing
callsites to use the new wrappers.

Refs #65342

Commit:
feat(profiling): add new use case for metrics dataset (#65152)

Use Case definition for `profiles`.

This is the first step to move us from the current custom `functions`
dataset for profile metrics to the `generic-metrics` platform

billyvg committed:
feat(replay): Add sdk version as tag to `replay.processing-error` metric (#65457)

narsaynorath committed:
fix(queries): Layout for Onboarding CTA (#65534)

Wrap the onboarding CTA with `ModuleLayout.Full` so it takes the full
space.

Before:
<img width="1435" alt="Screenshot 2024-02-21 at 10 52 28 AM"
src="https://github.com/getsentry/sentry/assets/22846452/564ea05b-3a26-4410-a676-de8229390a11">

After:
<img width="1491" alt="Screenshot 2024-02-21 at 10 52 17 AM"
src="https://github.com/getsentry/sentry/assets/22846452/a2e15d2a-011b-4a65-9488-a664a6f5bc89">

iamrajjoshi committed:
feat(organization settings): Always show audit log  tab (#65363)

* Always shows audit log tab in organization settings
* If user doesn't have access, it will show warning

User with access:
![Screenshot 2024-02-16 at 2 46
38 PM](https://github.com/getsentry/sentry/assets/33237075/3c73b235-9350-4839-831a-42d555c327eb)

User without access:
![Screenshot 2024-02-16 at 2 46
25 PM](https://github.com/getsentry/sentry/assets/33237075/e1e268e9-a6eb-41b7-8d6b-91cbdd796dbe)

Fixes
https://github.com/getsentry/team-core-product-foundations/issues/86

RyanSkonnord committed:
test(hc) Tag and fix OrganizationAndStaffPermissionTest (#65490)

Add `@region_silo_test`. Fix a failure in `test_staff_passes_2FA` by
setting up the `request.user` as an RpcUser, as it would be on a region
silo. This prevents a bug where a duck-typed call to `use_2fa()` is
executed on the `User` model class, which naturally tries to read
control silo tables.

RyanSkonnord committed:
test(hc): Tag ApiApplicationTest (#65489)

Assuming this is similar to other `@no_silo_test` cases in
`tests/acceptance/` and this one was just missed until now.

RyanSkonnord committed:
fix: Safely extract name attribute from Endpoint when dispatching (#65491)

Guard against an error that occurs when an Endpoint instance lacks a
`__name__` attribute. (Observed only in HC environments so far.)

edwardgou-sentry committed:
feat(webvitals): Adds performance score profiles for chrome, edge, and opera inp (#65254)

Adds performance score profiles for chrome, edge, and opera inp. Firefox
and Safari do not support inp yet.
For "Chrome INP", we allow the browser name condition to be either
"Chrome" or "Google Chrome"

mrduncan committed:
ref(issues): Mark OrganizationProjectsSentFirstEventEndpoint private (#65509)

This controls features in the Sentry UI and isn't particularly useful as
a public API.

cathteng committed:
chore(slack): log payload for unfurl error (#65543)

cathteng committed:
ref(slack): remove default tags (#65478)

JoshFerge committed:
ref(typing): type a few debug email endpoints (#65500)

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

snigdhas committed:
fix(issue-priority): Only evaluate high priority event condition for new or escalating issues (#65517)

This fixes a bug in the priority alert logic using the new
Group.priority value. We should only alert on high priority issue for
issues that escalate or are new.

JoshFerge committed:
fix(feedback): better metrics in create_feedback (#65494)

- [x] Adds better metrics for feedback creation, so we can ensure that a
loss of feedbacks can be noticed immediately

schew2381 committed:
chore(staff): Create staff and project permission class (#65482)

scttcper committed:
ref(ui): Switch AbstractIntegrationDetailedView to abstract (#65514)

scttcper committed:
ref(ui): Declare release & resource class context types (#65515)

JoshFerge reviewed (approved):
lgtm. what happens if a user searches by priority but they don't have the flag enabled?

JoshFerge committed:
ref(issues): move issues owned endpoints into issues.endpoint folder (#65359)

- Move individual endpoints owned by issues team into issues.endpoint
folder. This will help us with ownership rules, etc. More endpoints to
be moved, as well as helper functions in coming PRs.

---------

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

k-fish committed:
feat(metrics-extraction): Add warning for equations (#65552)

### Summary
Equations aren't supported by metrics extraction, this adds a warning
that indicates the equation won't work.

Closes #55990

#### Screenshot
![Screenshot 2024-02-21 at 12 49 22
PM](https://github.com/getsentry/sentry/assets/6111995/c3b993c3-e259-4f0e-bf4f-bc7932124911)

JoshFerge committed:
ref(replays): flatten event iterator (#65291)

re-opening @cmanallen's https://github.com/getsentry/sentry/pull/62151
with a couple of tweaks. will be useful for my upcoming changes to add
data from replay events here.

leeandher committed:
ref(auto-assignment): Remove group check for force-assignment (#65486)

Now that the only caller of force assignment has been updated to send
`force_autoassign` we no longer need the implicit assumption when no
`event` is present (see
https://github.com/getsentry/getsentry/pull/13006). Going forward, to
use this function for forcing auto-assignment, `force_autoassign` must
be set.

joshuarli committed:
fix(devenv): remove venv.sync (#65502)

JoshFerge committed:
feat(feedback): add project option for enabling crash/user report notifs (#65352)

- [x] add new epoch, epoch 12
- [x] create new project option
`sentry:feedback_user_report_notification`, and default it to True in
epoch 12.
- [x] add to project model / serializer

schew2381 committed:
chore(staff): Let staff access project users endpoint (#65483)

Requires https://github.com/getsentry/sentry/pull/65482

lobsterkatie committed:
fix(grouping): Fix bug in `assign_to_event` tests (#65190)

This fixes a bug in the `assign_event_to_group` tests, where it assumed that it was only the first call to `find_existing_grouphash`/`find_existing_grouphash_new` which mattered in terms of determining what was found. Though that's true given the current `_save_aggregate` logic, it won't be with the new logic.

This changes it so that:
- The results from `find_existing_grouphash` are unwrapped from their tuples first, so that the unwrapping happens for all results rather than just the first one.
- Any results which don't find anything are filtered out.

We can then safely take the first result, since our logic uses the first real result it finds.

snigdhas committed:
Merge branch 'master' into snigdha/priority-default-be
